[inductor] overlap scheduling nodes benchmarking logs and fix

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #169947


1. Fix: benchmark_collective_with_cuda_events_impl was benchmarking without convertion FakeTensor -> real
2. Adding logging to tlparse of runtime estimations 'benchmarked' vs 'analytical', will be logged only if 'benchmark' mode is on. (In this case overhead for analytical calculation is minimal).
3. As we do warmup=1, nruns=3, using median for stability, as warmup 1 can be not enough. median is more stable for outliers in this case.



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

IvanKobzarev committed:
[inductor] overlap scheduling nodes benchmarking logs and fix

[ghstack-poisoned]

eellison commented on torch/_inductor/fx_passes/node_runtime_estimation.py:
```diff
@@ -68,19 +74,33 @@ def _benchmark_collective_with_cuda_events_impl(
+        stride = [get_hint(s) for s in t.stride()]
+
+        if any(s is None for s in itertools.chain(shape, stride)):
+            raise ValueError("Cannot convert tensor with symbolic dimensions")
```
this shouldn't occur because can_benchmark_collective should have already checked unbacked.. maybe update error msg? / add comment

eellison reviewed (approved):
Looks good ! one small comment on code movement

IvanKobzarev commented on torch/_inductor/fx_passes/overlap_scheduling.py:
```diff
@@ -498,6 +546,61 @@ def _calculate_compute_node_domination_index(self) -> dict[fx.Node, int]:
 
         return domination_index
 
+    def _log_compute_estimations(
```
yes, will move all _log* functions there.

IvanKobzarev committed:
Update on "[inductor] overlap scheduling nodes benchmarking logs and fix"



1. Fix: benchmark_collective_with_cuda_events_impl was benchmarking without convertion FakeTensor -> real
2. Adding logging to tlparse of runtime estimations 'benchmarked' vs 'analytical', will be logged only if 'benchmark' mode is on. (In this case overhead for analytical calculation is minimal).
3. As we do warmup=1, nruns=3, using median for stability, as warmup 1 can be not enough. median is more stable for outliers in this case.



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]
